### ISSUE TITLE
[Bug]: mo can not restrat 

### ISSUE THREAD

**Issue by aressu1985** [matrixorigin/matrixone#6526]
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Environment

```Markdown
- Version or commit-id (e.g. v0.1.0 or 8b23a93):b9c225f4f44dcd62d6f577394d705b6aed55188e
- Hardware parameters:
- OS type:
- Others:
```

### Actual Behavior

2022/11/12 21:17:49.947045 +0800 ERROR rpc-client[hakeeper-client([])] morpc/client.go:334 gc inactive backends task stopped
2022/11/12 21:17:49.952223 +0800 INFO db/open.go:59 open-tae {"operation": "End", "operand": "open", "cost": "1m59.892578361s", "err": null}
panic: timeout

goroutine 124 [running]:
github.com/matrixorigin/matrixone/pkg/vm/engine/tae/logstore/driver/logservicedriver.newClient(0xc019987280, 0x10?, 0x22d60c0?)
        /data1/sudong/mo-regression/matrixone/pkg/vm/engine/tae/logstore/driver/logservicedriver/clientpool.go:51 +0x212
github.com/matrixorigin/matrixone/pkg/vm/engine/tae/logstore/driver/logservicedriver.(*clientpool).newClientFactory.func1()
        /data1/sudong/mo-regression/matrixone/pkg/vm/engine/tae/logstore/driver/logservicedriver/clientpool.go:103 +0x36
github.com/matrixorigin/matrixone/pkg/vm/engine/tae/logstore/driver/logservicedriver.newClientPool(0x64, 0x64, 0xc0006938f0)
        /data1/sudong/mo-regression/matrixone/pkg/vm/engine/tae/logstore/driver/logservicedriver/clientpool.go:95 +0x11f
github.com/matrixorigin/matrixone/pkg/vm/engine/tae/logstore/driver/logservicedriver.NewLogServiceDriver(0xc019987400)
        /data1/sudong/mo-regression/matrixone/pkg/vm/engine/tae/logstore/driver/logservicedriver/driver.go:79 +0x8e
github.com/matrixorigin/matrixone/pkg/vm/engine/tae/logstore/store.NewStoreWithLogserviceDriver(0xc019987280)
        /data1/sudong/mo-regression/matrixone/pkg/vm/engine/tae/logstore/store/store.go:51 +0xca
github.com/matrixorigin/matrixone/pkg/vm/engine/tae/wal.NewDriverWithLogservice(0xc00dea1620?)
        /data1/sudong/mo-regression/matrixone/pkg/vm/engine/tae/wal/driver.go:55 +0x19
github.com/matrixorigin/matrixone/pkg/vm/engine/tae/db.Open({0xc00b462140, 0x40}, 0x0?)
        /data1/sudong/mo-regression/matrixone/pkg/vm/engine/tae/db/open.go:90 +0x765
github.com/matrixorigin/matrixone/pkg/vm/engine/tae/rpc.openTAE({0xc00565a400, 0x3c}, 0x41ea47?)
        /data1/sudong/mo-regression/matrixone/pkg/vm/engine/tae/rpc/handle.go:589 +0x172
github.com/matrixorigin/matrixone/pkg/vm/engine/tae/rpc.NewTAEHandle({0xc00565a400?, 0x0?}, 0xc019987280?)
        /data1/sudong/mo-regression/matrixone/pkg/vm/engine/tae/rpc/handle.go:66 +0x39
github.com/matrixorigin/matrixone/pkg/txn/storage/tae.NewTAEStorage({0xc00565a400, 0x3c}, {{0x2, 0x1, {}, {0x0, 0x0, 0x0}, 0x0}, 0x40002, ...}, ...)
        /data1/sudong/mo-regression/matrixone/pkg/txn/storage/tae/storage.go:56 +0x1af
github.com/matrixorigin/matrixone/pkg/dnservice.(*store).newTAEStorage(0xc0031f0780, {{0x2, 0x1, {}, {0x0, 0x0, 0x0}, 0x0}, 0x40002, {0xc0058f2100, ...}, ...}, ...)
        /data1/sudong/mo-regression/matrixone/pkg/dnservice/factory.go:145 +0x15f
github.com/matrixorigin/matrixone/pkg/dnservice.(*store).createTxnStorage(0xc0031f0780, {{0x2, 0x1, {}, {0x0, 0x0, 0x0}, 0x0}, 0x40002, {0xc0058f2100, ...}, ...})
        /data1/sudong/mo-regression/matrixone/pkg/dnservice/factory.go:73 +0x1ff

### Additional information

[server.norestart.log.zip](https://github.com/matrixorigin/matrixone/files/9995075/server.norestart.log.zip)

**Comment by volgariver6** [comment]
From the mo log, there are 267634 log entries need to replay, which takes long time and cause a timeout panic in tae.

The root cause may be the truncation request from tae to logservice delays too long.

**Comment by volgariver6** [comment]
About raft in dragonboat and logservice:

1. When restart, raft leader cannot be elected until applied index catch up committed index.
2. The catching up of applied index is automatically done by replaying logs after the snapshot is applied.
3. The advance of applied index is only related to logservice, and has nothing to do with tae and DN.

**Comment by volgariver6** [comment]
Have done some tests in my local env. The result:

1. It took 6s to finish the log replaying for 10869 indexes and 61 log files, each log file is 64MB.
2. There are 260000+ indexes and 1570 log files need to replay in @aressu1985 's test env. The number of index and log files are 25 times than my local env. So the estimated time is 6s*25=150s. Maybe it was normal to take that long to complete and a timeout panic happened.

The following shoule be done:

1. Optimize the performance of replay log.
2. Modify the system startup judgment conditions, and only after the leader is elected can the hakeeper start successfully.
3. Frequency optimization of truncate request sent by tae.
4. When tae reads log entries, it reads 2M each time, which can be expanded to 64M.

**Comment by volgariver6** [comment]
There is a problem that the size of record tae append to logservice is enlarged. And it has been fixed by #6555.

**Comment by volgariver6** [comment]
fixed by #6701 in main branch and #6702 in 0.6.0 branch.

**Comment by aressu1985** [comment]
will test this issue tomorrow

**Comment by aressu1985** [comment]
fixed:8f3aa6a337b415b7f99339dcc1f730c8c165ef40
